enterprise/frontend/src/embedding-sdk/components/public/InteractiveDashboard/EditableDashboard.tsx

```diff
@@ -209,19 +222,32 @@ export const EditableDashboard = ({
           })
         }
       >
-        {adhocQuestionUrl ? (
-          <InteractiveAdHocQuestion
-            questionPath={adhocQuestionUrl}
-            onNavigateBack={onNavigateBackToDashboard}
-            {...drillThroughQuestionProps}
-          />
-        ) : (
-          <EditableDashboardInner
-            drillThroughQuestionProps={drillThroughQuestionProps}
-            onEditQuestion={onEditQuestion}
-          />
-        )}
+        {match(finalRenderMode)
```

Comment:
Use an enum instead. Since a boolean-like variable isn't cutting it anymore, we have to render 3 types of UI here.